install/github/github.go

```diff
@@ -19,181 +19,44 @@ package github
 import (
```

Comment:
See https://github.com/ossf/scorecard-action/pull/833#issuecomment-1222175743:

> In the next batch of updates, I'd like to combine the following GitHub clients:
> 
>   * https://github.com/ossf/scorecard-action/blob/main/github/github.go
>   * https://github.com/ossf/scorecard-action/blob/main/install/github/github.go
> 
> We currently have the following comment in code:
> 
> ```go
> // ParseFromURL is a function to get the repository information.
> // It is decided to not use the golang GitHub library because of the
> // dependency on the github.com/google/go-github/github library
> // which will in turn require other dependencies.
> ```
> 
> ...but as we've introduced the `go-github` library to support the installer tool, it makes sense to deduplicate common code paths.
> 
> As part of that deduplication, I'll be able to mock the GitHub client and write a few tests for previously uncovered code paths.